lang/LangSource/PyrSignal.cpp

```diff
@@ -537,7 +537,7 @@ bool signal_equal_xx(VMGlobals *g, PyrObject* ina, PyrObject* inb)
 	float *endptr = a + ina->size;
 	if (ina->size != inb->size) return false;
 	if (slotRawSymbol(&ina->slots[ kSignalRate ]) != slotRawSymbol(&inb->slots[ kSignalRate ])) return false;
-	while (a < endptr) { if (*a++ != *b++) return false; }
+	while (a < endptr) { if (*++a != *++b) return false; }
```

Comment:
this should be refactored for clarity:

```
while (a < endptr) {
    a++;
    b++;
    if (*a != *b) {
        return false;
    }
}
```

Reply:
actually, let's keep this as clear as possible:

```
bool signal_equal_xx(VMGlobals *g, PyrObject* ina, PyrObject* inb)
{
	float *a = (float*)(ina->slots);
	float *b = (float*)(inb->slots);
	if (ina->size != inb->size) return false;
	if (slotRawSymbol(&ina->slots[ kSignalRate ]) != slotRawSymbol(&inb->slots[ kSignalRate ])) return false;
	for (int i = 0; i < ina->size; i++) {
		if (a[i] != b[i]) {
			return false;
		}
	}
	return true;
}
```

Reply:
I like using `std::equal`, and I would expect a performance bump since that's probably doing some simd optimization - but I would suggest merging the simple fix first, and then refactoring (with std::equal or whatever else) as a separate change.
Also, it's worth checking wether nova_simd has a wrapper for simd equality comparison - that could be the best thing to use here.